### PR TITLE
Ignore `total_shards_per_node` setting on searchable snapshots in frozen

### PR DESCRIPTION
**Pull request by gmarouli**
frozen

This PR is an alternative way of addressing #90871, instead of allowing the allocate settings in frozen and burdening the user to have to remember that the `total_shards_per_node` was set, we decided to unset this property via ILM. As discussed here: https://github.com/elastic/elasticsearch/issues/90871#issuecomment-1596632805.

This PR adds the `index.routing.allocation.total_shards_per_node` to the `MountSearchableSnapshotRequest` as an ignored setting.

Food for thought, should we do the same for cold?

### CODE REVIEW

**Review by andreidan — APPROVED** [review]
LGTM, thanks for fixing this Mary

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/MountSnapshotStep.java [hunk 1]
```diff
@@ -184,6 +179,24 @@ static Optional<String> overrideTierPreference(String phase) {
...
+     * time, make it's way forward from _that_ step forward in the ILM policy. We'll re-set this setting on the restored index at a later
+     * step once we restored a deterministic execution state
+     * - index.routing.allocation.total_shards_per_node: It is  likely that frozen tier has fewer nodes than the hot tier.
+     * Keeping this setting runs the risk that we will not have enough nodes to allocate all the shards in the
+     * frozen tier and the user does not have any way of fixing this. For this reason, we ignore this setting when moving to frozen.
+     */
+    static String[] ignoredIndexSettings(String phase) {
+        // if we are mounting a searchable snapshot in the hot phase, then we should not change the total_shards_per_node setting
+        if (TimeseriesLifecycleType.FROZEN_PHASE.equals(phase)) {
+            return new String[] { LifecycleSettings.LIFECYCLE_NAME, "index.routing.allocation.total_shards_per_node" };
```
[the remark this answers is not in the document]
Comment (gmarouli) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/MountSnapshotStep.java:195]: Definitely, let me check.

**Comment by gmarouli** [comment]
> Food for thought, should we do the same for cold?

After some consideration, we decided **not for now**. There reasons are:

- there’s already a workaround that appears to be good enough
- we are concerned it might be a surprising change of behaviour for some users who only have fully mounted indices in cold (and no allocate action).

We understand that maybe the fact that the workaround exists might be hiding the need for this, that's why the answer is not for now.

**Comment by gmarouli** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.9|[](https://github.com/elastic/elasticsearch/pull/98002)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**